src/features/common/Layout/Navbar/microComponents/NavbarItems.tsx

```diff
@@ -0,0 +1,40 @@
+import type { HeaderItem, NavbarItemHeaderKey } from '../defaultTenantConfig';
+
+import { useTenant } from '../../TenantContext';
+import UserProfileButton from './UserProfileButton';
+import { useState } from 'react';
+import NavbarItemGroup from './NavbarItemGroup';
+
+const NavbarItems = () => {
+  const { tenantConfig } = useTenant();
+  const headerItems = tenantConfig.config.header.items;
+  const [openMenuKey, setOpenMenuKey] = useState<NavbarItemHeaderKey | null>(
+    null
+  );
+
+  const renderHeaderItem = (navItem: HeaderItem) => {
+    return navItem.headerKey === 'me' ? (
+      <div className="profileButtonContainer" key={navItem.headerKey}>
+        <UserProfileButton />
+      </div>
+    ) : (
+      <NavbarItemGroup
+        navItem={navItem}
+        openMenuKey={openMenuKey}
+        setOpenMenuKey={setOpenMenuKey}
+      />
+    );
+  };
+
+  return headerItems.length > 0 ? (
+    <nav className={'headerItems'}>
```

Comment:
Shouldn't we first apply filters and then render the `nav` element? What if all headerItems are set to visible: false?

Reply:
is this scenario possible ?